dags/app/user_profile/udf.py

```diff
@@ -0,0 +1,318 @@
+    table_id = f"{BIGQUERY_PROJECT}.{BIGQUERY_DATASET}.kktix_ticket_{task_type}_updates"  # 要建立的暫存表
+
+    job_config = bigquery.LoadJobConfig(
+        write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,  # 覆蓋舊表，如果表不存在則建立
+        schema=[
+            bigquery.SchemaField(task_type, "STRING"),
+            bigquery.SchemaField("category", "INT64"),
+        ],
+    )
+    job = client.load_table_from_dataframe(df, table_id, job_config=job_config)
+    job.result()
+
+
+def chunk_data(data, batch_size) -> Generator[list[str], None, None]:
+    for i in range(0, len(data), batch_size):
+        yield data[i : i + batch_size]
+
+
+def call_gemini(model: str, max_output_tokens_on_model: int, prompt: str) -> str | None:
+    client = genai.Client(api_key=check_gemini_api_key())
+    response = client.models.generate_content(
+        model=model,
+        contents=[prompt],
+        config=types.GenerateContentConfig(
+            max_output_tokens=max_output_tokens_on_model
+        ),
+    )
+    return response.text
+
+
+def process_table(
+    model: str, max_output_tokens_on_model: int, batch_size: int, task_type: str
+) -> None:
+    """
+    讀取檔案，根據任務類型處理內容，並將結果寫入 CSV 檔案。
+
+    Args:
+        output_filename: 輸出 CSV 檔案的名稱。
+        model: 要使用的 Gemini 模型名稱。
+        max_output_tokens_on_model: 模型允許的最大輸出 token 數。
+        batch_size: 每次處理的記錄數。
+        task_type: 任務類型 ('organization' 或 'job_title')。
+    """
+    # Read the input file
+    data_list = read_kktix_ticket_user_profile(task_type)
+    results_to_update = []
+
+    logging.info(
+        f"共讀取到 {len(data_list)} 筆記錄，將分成約 {len(data_list) // batch_size + (1 if len(data_list) % batch_size > 0 else 0)} 個批次處理..."
+    )
+
+    # Process in batches
+    for index, chunk in enumerate(chunk_data(data_list, batch_size), start=1):
+        batch_text = json.dumps(chunk, ensure_ascii=False)
+        batch_prompt = get_task_config(task_type) + "\n資料列表:\n" + batch_text
+
+        logging.info(f"正在處理批次 {index}...")
+
+        # Check batch size against model limit (optional, but good practice)
+        # Note: Token count is not a simple character count, this is a rough check
+        if len(batch_text) > max_output_tokens_on_model * 0.8:  # Use a buffer
+            logging.warning(
+                f"警告：批次 {index} 的大小可能接近或超過模型的最大輸入限制 ({max_output_tokens_on_model} tokens)。"
+            )
+
+        try:
+            response = call_gemini(model, max_output_tokens_on_model, batch_prompt)
+            if response:
+                try:
+                    clean_text = response.strip()
+                    clean_text = clean_text[len("```json") :].strip()
+                    clean_text = clean_text[:-3].strip()
+                    result_json = json.loads(clean_text)
+                except json.JSONDecodeError:
+                    raise ValueError(f"批次 {index} JSON解析錯誤：{clean_text}")
+            results_to_update.extend(result_json)
+        except Exception as e:
```

Comment:
It would be nice if we could catch only the exceptions that might be raised in the try block